lib/cretonne/src/ir/types.rs

```diff
@@ -35,6 +35,22 @@ pub const VOID: Type = Type(0);
 // 512-bit SIMD vectors.
 include!(concat!(env!("OUT_DIR"), "/types.rs"));
 
+/// Possible derivation functions for types. Used in runtime typechecking
+pub enum DerivedFunctions {
```

Comment:
Are you sure this (and the `transform` function below) is necessary?

It seems unlikely that we would need to apply type functions that are unknown at compile time.